Branch-SDK/src/io/branch/referral/ServerRequest.java

```diff
@@ -279,5 +279,26 @@ protected boolean doesAppHasInternetPermission(Context context){
 		return result == PackageManager.PERMISSION_GRANTED;
 	}
 
+	/**
+	 * Updates the google ads parameters. This should be called only from a background thread since it involves GADS method invocation using reflection
+	 *
+	 * @param sysObserver {@link SystemObserver} instance.
+	 */
+	public void updateGAdsParams(SystemObserver sysObserver) {
+		try {
+			String advertisingId = sysObserver.getAdvertisingId();
+			if (advertisingId != null && getPost() != null) {
+				getPost().put(Defines.Jsonkey.GoogleAdvertisingID.getKey(), advertisingId);
```

Comment:
We are actually not reading from the Json Object. Only writing to the Post object. Also "ServerRequestObject" is processed only by a single thread at a  time. Is there any other chance for concurrent exception?